### DISCUSSION TITLE
TripsLayer change width dynamically

### DISCUSSION THREAD

**Discussion by BenjaminSAILcloud** [visgl/deck.gl#5770] [Q&A] [answered]
Hi,

![image](https://user-images.githubusercontent.com/33052668/118012778-b4aea780-b351-11eb-8aaf-e6c04334413d.png)

I am trying to make one of the paths larger than the others. When using the `getWidth` with a number, the width is changed directly. But it changes the width for all paths. I want to use `getWidth` with a function where I could something like this :
```ts
new TripsLayer({
    ...
    getWidth: (d) => {
        return selectedTrails.includes(d.id) ? 15 : 4;
    },
    ...
})
```
I believe the function is called once and never again. If I define `selectedTrails` with ids it works, but if `selectedTrails` changes after that, it does not change dynamically.
Is there a way to force the Layer to udpate whenever the `selectedTrails` updates ?

**Comment by Pessimistress** [comment, marked as answer]
[Why Doesn't My Layer Update with New Props?](https://deck.gl/docs/developer-guide/using-layers#why-doesnt-my-layer-update-with-new-props)

**Comment by BenjaminSAILcloud** [comment]
### Solution : 
```diff
new TripsLayer({
    ...
    getWidth: (d) => {
        return selectedTrails.includes(d.id) ? 15 : 4;
    },
+   updateTriggers: {
+       getWidth: [selectedTrails],
+   },
    ...
})
```

Thanks @Pessimistress
